core/templates/services/search.service.ts

```diff
@@ -16,215 +16,234 @@
  * @fileoverview search service for activityTilesInfinityGrid
  */
 
-require('services/services.constants.ajs.ts');
-
-import { EventEmitter } from '@angular/core';
-
-angular.module('oppia').factory('SearchService', [
-  '$http', '$log', '$translate', 'SEARCH_DATA_URL',
-  function($http, $log, $translate, SEARCH_DATA_URL) {
-    var _lastQuery = null;
-    var _lastSelectedCategories = {};
-    var _lastSelectedLanguageCodes = {};
-    var _searchCursor = null;
-    var _isCurrentlyFetchingResults = false;
-    var numSearchesInProgress = 0;
-    var _searchBarLoadedEventEmitter = new EventEmitter();
-    var _initialSearchResultsLoadedEventEmitter = new EventEmitter();
-    // Appends a suffix to the query describing allowed category and language
-    // codes to filter on.
-    var _getSuffixForQuery =
-      function(selectedCategories, selectedLanguageCodes) {
-        var querySuffix = '';
-
-        var _categories = '';
-        for (var key in selectedCategories) {
-          if (selectedCategories[key]) {
-            if (_categories) {
-              _categories += '" OR "';
-            }
-            _categories += key;
-          }
-        }
-        if (_categories) {
-          querySuffix += '&category=("' + _categories + '")';
-        }
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { Injectable, EventEmitter } from '@angular/core';
+
+import { ServicesConstants } from 'services/services.constants';
+import { ExplorationSummaryBackendDict } from 'domain/summary/exploration-summary-backend-api.service';
+import { SearchBackendApiService, SelectionDetails, SelectionList } from './search-backend-api.service';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class SearchService {
+  private lastQuery: string = null;
+  private lastSelectedCategories: SelectionList = {};
+  private lastSelectedLanguageCodes: SelectionList = {};
+  private searchCursor: string = null;
+  private isCurrentlyFetchingResults = false;
+  private searchBarLoadedEventEmitter = new EventEmitter<string>();
+  private initialSearchResultsLoadedEventEmitter =
+    new EventEmitter<ExplorationSummaryBackendDict>();
+  public numSearchesInProgress = 0;
+
+  constructor(
+    private searchBackendApiService: SearchBackendApiService) {
+  }
 
-        var _languageCodes = '';
-        for (var key in selectedLanguageCodes) {
-          if (selectedLanguageCodes[key]) {
-            if (_languageCodes) {
-              _languageCodes += '" OR "';
-            }
-            _languageCodes += key;
-          }
+  private getSuffixForQuery(
```

Comment:
We add `_` in front of private variables and methods. Can you change that here and also to the private stuff above.
```suggestion
  private _getSuffixForQuery(
```

Reply:
done